### ISSUE TITLE
Add right click "search with google" action

### ISSUE THREAD

**Issue by RAnders00** [Chatterino/chatterino2#2058]
**What should be added?**
When right-clicking on some selected text, there should be a "Search with google..." option that opens the browser with a google search for the selected phrase.

**Why should it be added?**
Can be useful to quickly google things other chatters are talking about.

This is also a feature in google chrome and other modern browsers, I personally use it quite often in the browser and I imagine it could be similarly useful in Chatterino. I've also seen "Search with google..." being an option in the Intellij IDEA console where it can be used to quickly google for error messages etc.

**Comment by TranRed** [comment]
maybe we could even offer a advanced setting for alternate search engines for people who dislike google

**Comment by leon-richardt** [comment]
> advanced setting for alternate search engines

I suggest such setting to be located on the "External Tools" settings page.

**Comment by Y0dax** [comment]
What alternate search engines do we think make sense here? 

From popularity: 
Bing?
Yahoo?

**Comment by leon-richardt** [comment]
> What alternate search engines do we think make sense here?

I believe the idea here is to support custom query schemas, similar to what to the "Custom stream player" and "Image Uploader" section in the "External tools" settings.

**Comment by TranRed** [comment]
I was thinking about duckduckgo. I think this is popular with the "no google"-demographic

**Comment by TranRed** [comment]
Also was Leon said.

**Comment by ALazyMeme** [comment]
I think a better solution would to somehow have the ability to use commands in a right click function on a message. From there, a user can make a `/openurl` command and use whatever search engine they want.

**Comment by remd** [comment]
#2067 provided an implementation of this feature but was closed until discussion on this issue was finished, however discussion seems to have stalled out without any significant contention between participants.

I humbly ask that #2067 be resurrected and the discussion here continues as needed.

**Comment by ALazyMeme** [comment]
> #2067 provided an implementation of this feature but was closed until discussion on this issue was finished, however discussion seems to have stalled out without any significant contention between participants.
> 
> I humbly ask that #2067 be resurrected and the discussion here continues as needed.

What other discussion would you like? Seems like my suggestion would offer the right amount of customization, but wouldn't specifically fall under this issue.

**Comment by RAnders00** [comment]
I don‘t like alazymeme‘s suggestion. Seems like using commands like that would result in a  bad user experience for a simple feature that would offer a better user experience if implemented explicitly. (It would allow the UI to be custom-fit for the task for example, instead of repurposing the commands for that)

**Comment by LosFarmosCTL** [comment]
I would really like the option to add custom context menu entries that launch commands, but I kind of agree that there are UX issues with using commands for this specific feature:

1. Commands are in general not really discoverable in chatterino at the moment, i would suspect that probably not even 5% of all chatterino users use or even know of the /openurl command, so just adding this possibility via commands probably means that most people are never going to even know that it is possible.
2. The UI for setting this up would not be very intuitive, a lot of people are already overwhelmed by chatterinos current command system and don’t really know much about it except using it to spam copypastas.

There are imo upsides to this as well though, using commands would make this really customizable for the people that do want to do so (i.e. setting whatever other search engine you want, adding multiple entries for different search engines or website searches, etc.)

I could see this being implemented sensible using commands, but there would have to be some effort to solve the 2 big UX issues, maybe exploring the possibility of chatterino providing some built-in example commands, comparable to how there are default highlights as well. This would imo in general be a good idea, since it could help a lot with discoverability to just kinda show off command features a bit, because realistically next to no one will ever look at something like the wiki article explaining them, but just seeing a few examples could show people whats possible.

In the end those might just be different features that shouldnt be merged together, and it could make sense to just implement some search functionality with a few selectable search engines and maybe just additionally have the possibility of adding commands to the context menu.

**Comment by remd** [comment]
> What other discussion would you like? Seems like my suggestion would offer the right amount of customization, but wouldn't specifically fall under this issue.

#2067 was closed because a desire for certain people to provide feedback before moving forward was expressed. That feedback was never obtained, and I don't know that it's necessary to ping those people directly. In my opinion, #2067 is worth merging as-is. Whatever discussion is needed to get #2067 moving again is what I was encouraging.

Your alternative suggestion works for me. I have no problem configuring my own commands to fire off a right-click menu option.

However, I do agree with the sentiment that the user experience for this particular feature would likely be served better by an explicit implementation as mentioned before.

**Comment by kornes** [comment]
Support the idea with proper customization like #3603 suggests. Hardcoding search engine doesn't sound right to me, because some prefer their local services (Baidu in china, Yandex in russia), others prefer privacy first Duckduckgo. 
Adding that to settings will also allow other use cases: translators, wikipedia, github search, whatever given user finds useful.

**Comment by Felanbird** [comment]
With the merging of #3738, this feature now technically exists.

Custom search commands can be setup and accessed via the new right-click execute command feature.

![image](https://user-images.githubusercontent.com/41973452/168464779-1f1ed8fd-1a20-435e-ae1a-df3ddec3eabc.png)

Right click with no highlighting will search the full message text, highlighting will only search the highlighted words.

// google
`/openurl https://google.com/search?q={1+}`

// bing
`/openurl https://www.bing.com/search?q={1+}`

// startpage
`/openurl https://www.startpage.com/sp/search?query={1+}`

// duckduckgo
`/openurl https://duckduckgo.com/?q={1+}`

**Comment by RAnders00** [comment]
I‘d like to put in a word to reopen this issue. I‘m not a fan of the user experience we’d be offering if we left it at that.

The average chatterino user or streamer is not technically versed enough to set this up. And the fact that you even have to set it up in the first place is problematic in my opinion as well.

**Comment by kornes** [comment]
agree, we should add some of the examples from Felanbird as default commands in settings. 
That will require adding support to toggle commands on/off tho, so all defaults can start disabled.

**Comment by 8thony** [comment]
- #6743

added support for search in the right-click menu. (disabled by default)
You can choose Google, Bing, or DuckDuckGo as a preset or fill in your own preferred search engine.
